src/puya/ir/builder/blocks.py

```diff
@@ -41,18 +42,23 @@ def __init__(
         self.ssa = BraunSSA(blocks, parameters, self.active_block)
         self.ssa.seal_block(self.active_block)
 
-    @property
-    def blocks(self) -> Sequence[BasicBlock]:
-        return self._blocks
+        self._pending_labelled_blocks = dict[awst_nodes.Label, BasicBlock]()
+        self._created_labelled_blocks = dict[awst_nodes.Label, BasicBlock]()
 
     @property
     def active_block(self) -> BasicBlock:
         return self._blocks[-1]
 
     def add(self, op: Op) -> None:
         """Add an op"""
-        curr = self._get_current_for_addition(op.source_location)
-        if curr is not None:
+        curr = self.active_block
+        if curr.terminated:
+            logger.error(
+                "unreachable code",
+                # function kinda sucks as a fallback, but it'll do for now
+                location=op.source_location or self._default_source_location,
```

Comment:
could maybe try `curr.terminator.source_location` before the function default as it would hopefully be closer to the actual problem

Reply:
This was copied from existing code but that's a good point, I'll update this and the original code.